### PR TITLE
Edit Add Command

### LINKED ISSUE #58: Edit Add Command
Making all other fields optional except for name and phone number

### PR DESCRIPTION
**Pull request by BuffWuff1712**
Fixes #58

### CODE REVIEW

**Review by Vatinius — COMMENTED** [review]
Is changing the error message for the Add command intended as well?

The add error message still does not show that email onwards is optional
image

**Review by Vatinius — CHANGES_REQUESTED** [review]
Set my review to "Request changes"

**Review by Vatinius — COMMENTED** [review]
Need to make mental note if we want to stick with this formatting/styling of the empty email, address and tags field or not when we add individuals without filling up all the fields.

image
image

**Comment by BuffWuff1712** [comment]
Made changes to formatting of error and success messages for add command

**Review by Tkaixiang — APPROVED** [review]
tested - lgtm

**Review by Vatinius — APPROVED** [review]
lgtm!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -89,6 +93,10 @@ public static Address parseAddress(String address) throws ParseException {
     public static Email parseEmail(String email) throws ParseException {
         requireNonNull(email);
         String trimmedEmail = email.trim();
+        if (trimmedEmail.isEmpty()) {
```
Comment (sayomaki) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:96]: Ditto here.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -74,6 +74,10 @@ public static Phone parsePhone(String phone) throws ParseException {
     public static Address parseAddress(String address) throws ParseException {
         requireNonNull(address);
         String trimmedAddress = address.trim();
+        if (trimmedAddress.isEmpty()) {
```
Comment (sayomaki) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:77]: Do we need the check here? Since `Address.isValidAddress` has already been updated.

**Review by sayomaki — COMMENTED** [review]
I have looked through the code (despite it being merged), some comments (for future updates)

File: src/test/java/seedu/address/model/person/AddressTest.java [hunk 3]
```diff
@@ -13,20 +13,11 @@ public void constructor_null_throwsNullPointerException() {
...
-    }
-
     @Test
     public void isValidAddress() {
         // null address
         assertThrows(NullPointerException.class, () -> Address.isValidAddress(null));
 
-        // invalid addresses
-        assertFalse(Address.isValidAddress("")); // empty string
-        assertFalse(Address.isValidAddress(" ")); // spaces only
```
Comment (sayomaki) [inline src/test/java/seedu/address/model/person/AddressTest.java:29]: Maybe for this we can instead keep the test cases and just change the assertion from false to true?

**Pull request merged**